7-0-java-concurrency/7-0-0-hello-threads/src/test/java/HelloThreadsTest.java

```diff
@@ -0,0 +1,107 @@
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.*;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class HelloThreadsTest {
+
+    private Queue<Integer> concurrentLinkedQueue;
+
+    @BeforeEach
+    void setup() {
+        concurrentLinkedQueue = new ConcurrentLinkedQueue<>();
+    }
+
+    @SneakyThrows
+    @Test
+    @Order(1)
+    void createThread() {
+        Thread runningThread = HelloThreads.createThread(() -> concurrentLinkedQueue.add(5));
+        assertEquals(0, concurrentLinkedQueue.size());
+
+        runningThread.start();
+        runningThread.join();
+
+        verifyContentQueue();
+    }
+
+    @SneakyThrows
+    @Test
+    @Order(2)
+    void startThread() {
+        var thread = new Thread(() -> concurrentLinkedQueue.add(5));
+        HelloThreads.startThread(thread);
+        thread.join();
+
+        verifyContentQueue();
+    }
+
+    @SneakyThrows
+    @Test
+    @Order(3)
+    void getThreadName() {
+        var thread = new Thread(() -> concurrentLinkedQueue.add(5), "name");
+        var threadName = HelloThreads.getThreadName(thread);
+        thread.join();
+        assertEquals("name", threadName);
+
+        verifyContentQueue();
```

Comment:
It should not assert the queue content, since this method is only created to the thread name.